### PR TITLE
UID2-860 Add rename key functionality to admin panel

### PR DESCRIPTION
**Pull request by cYKatherine**
image

The above will be calling the backend with the below payloads:
image

### CODE REVIEW

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java [hunk 1]
```diff
@@ -431,6 +437,39 @@ private void handleClientRoles(RoutingContext rc) {
         }
     }
 
+    private void handleClientRename(RoutingContext rc) {
```
Comment (vishalegbert-ttd) [inline src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:440]: Can you add some tests to ClientKeyServiceTest? Should check if it works and should also check if it prevents a user from renaming to a non-unique name

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java [hunk 2]
```diff
@@ -431,6 +437,39 @@ private void handleClientRoles(RoutingContext rc) {
...
+            existingClient.withNameAndContact(newName);
+
+            List<ClientKey> clients = this.clientKeyProvider.getAll()
+                    .stream().sorted((a, b) -> (int) (a.getCreated() - b.getCreated()))
+                    .collect(Collectors.toList());
+
+            // upload to storage
+            storeWriter.upload(clients, null);
+
+            createPublisherSiteKeyIfNoneExists(existingClient);
```
Comment (vishalegbert-ttd) [inline src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:464]: Do we need this?

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java (hunk 1 above)
Comment (cYKatherine) [reply src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java:440]: Done

File: src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java (hunk 2 above)
Comment (cYKatherine) [reply src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java]: Removed

**Review by vishalegbert-ttd — APPROVED** [review]

**Pull request merged**